src/components/views/settings/Notifications.tsx

```diff
@@ -469,6 +470,25 @@ export default class Notifications extends React.PureComponent<IProps, IState> {
         });
     };
 
+    private validateKeyword = withValidation({
+        description: () => undefined,
+        hideDescriptionIfValid: true,
+        rules: [
+            {
+                key: "keyword_starts_with_period",
+                test(this: Notifications, { value }) {
+                    return !value || !/^\./.test(value);
+                },
+                invalid: () => _t("Keywords cannot start with '.'"),
+            },
+            {
+                key: "keyword_contains_fslash",
+                test: ({ value }) => !value || !/\//.test(value),
+                invalid: () => _t("Keywords cannot contain '/'"),
+            },
+        ],
```

Comment:
Where are these restrictions defined in the spec?

Reply:
I couldn't find these in the spec, though not sure I've looked in the right place. I based these rules on the synapse validation https://github.com/matrix-org/synapse/blob/acda9f07c858902ab072d49f6ad3e26df874d06e/synapse/rest/client/push_rule.py#L76-L90

Reply:
That validation is for the rule_id, not for the keyword itself, so it only breaks because Element uses the keyword as the rule_id too. The spec only says that room-specific and user-specific rules must be the room id or user id respectively. So the right solution here is to not use rule_id in a way which causes keywords to be more restrictive I would think.